bioio_imageio/reader.py

```diff
@@ -522,30 +523,58 @@ def _read_immediate(self) -> xr.DataArray:
             if image_length == 1:
                 image_data = reader.get_data(0)
 
-            # Handle many image formats like gif, mp4, etc
+            # Handle many-image formats like gif, mp4, etc
             elif image_length > 1:
-                # Read and stack all frames
-                frames = []
-                for frame in reader:
-                    frames.append(frame)
+                # 1) Read all frames into a list of numpy arrays
+                frames = [np.asarray(frame) for frame in reader]
+
+                # 2) Compute the maximum shape across all frames
+                all_shapes = [f.shape for f in frames]
+                max_ndim = max(len(s) for s in all_shapes)
+                max_shape = tuple(
+                    max((s[i] if i < len(s) else 0) for s in all_shapes)
+                    for i in range(max_ndim)
+                )
 
-                image_data = np.stack(frames)
+                # 3) Pad each frame up to max_shape
+                padded = []
+                for f in frames:
+                    pad_width = []
+                    for axis, target in enumerate(max_shape):
+                        current = f.shape[axis] if axis < f.ndim else 0
+                        pad_after = target - current
+                        pad_width.append((0, pad_after))
+                    f_padded = np.pad(
```

Comment:
Can you add a comment here describing why this is necessary

Reply:
awesome, merging this code will fix the other PR #11